### PR TITLE
Feat: Add React Native Wallet 0.68.5

### PR DESCRIPTION
**Pull request by crypblizz8**
https://user-images.githubusercontent.com/45455218/216589748-a8ee601b-631e-4c12-931b-786e662089d7.mov

In this sprint / PR I focused on
- Addresses 1/2 of #5.  (React Native 0.68.5)
- Screens: Onboarding / Home Screen + all the approval + eth methods modals
- WC Copy and Paste TextInput

Next Sprint
- Add in QR

Next Next Sprint
- WC QR Code
- Dark Mode
- (Scrolling Animation / Images)
- Figure out Shadows on React Native for the ActionButtons (iOS) Android elevation works
- Add in the details / deletion of specific "sessions" / Apps.
- Silence the MaxEventListners (From the W3WSDK emitting events)
- Cosmos + Solana + Tron
- Fonts for Android

Context for these above to be done later is / priorities:
1) This code base is getting quite large and I would like to get a review on this first so we have something we can show other developers what can be possible
2) After this PR is approved/merged, will need to switch to a `react-native-dapp` type of implementation for V2
3) Then can revisit `Next Sprint` + then come back to `next next sprint`

Considerations during PR Review
- Please provide opinion if the mneumonic storage process I do works
- Suggestive refactors for the future for the large amount of functions on the HomeScreen
- Want to get this out asap and then revisit with QR Code + some other QoL as we have to move onto other RN implementations

P.S.
For iOS: it was an easy: `npx react-native run-ios --simulator="iPhone 14 Pro Max"`
For Android: `npx react-native run-android` had some build failure problems and did a `./gradlew clean` in the`/android` folder. Also ensure you have not missed any setup steps such as export in your bash profile. [Docs](https://reactnative.dev/docs/environment-setup)

Feel free to reach out about RN setups 👍

### CODE REVIEW

File: wallets/rn_cli_wallet/src/utils/SignClient.ts [hunk 1]
```diff
@@ -1,14 +1,18 @@
-import SignClient from '@walletconnect/sign-client';
+import { Core } from '@walletconnect/core'
```
Comment (crypblizz8) [inline wallets/rn_cli_wallet/src/utils/SignClient.ts]: Todo: Tom --> Revert these changes

**Comment by bkrem** [comment]
Seeing Mac `.DS_Store` dotfiles being indexed in multiple dirs. Fix is adding `.DS_Store` to the .gitignore

File: README.md [hunk 2]
```diff
@@ -12,6 +12,28 @@ Catalogue of wallet and dapp examples WalletConnect's various SDKs via React Nat
 - Read through the various README files for further information
 - Submit any issues / feature requests.
 
+Note:
+If you are switching between all of these wallets i.e. `rn_cli_068_5` & `rn_cli_070_6`, recommend clearing cache and using this one command block.
```
Comment (bkrem) [inline README.md]:
```suggestion
If you are switching between these wallets (i.e. `rn_cli_068_5` & `rn_cli_070_6`), it is recommended to clear caches using the following command:
```

File: wallets/rn_cli_wallet_068_5/src/components/CircleActionButton.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import {TouchableOpacity, StyleSheet, Image} from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+
+interface CircleActionButtonProps {
+  copyImage: boolean;
+  handlePress: () => void;
+}
+
+const clipboardImage = require('../assets/Copy.png');
+const QRCodeImage = require('../assets/QRCode.png');
```
Comment (bkrem) [inline wallets/rn_cli_wallet_068_5/src/components/CircleActionButton.tsx:11]: Is there a specific reason for using `require` here? E.g. limitations in RN around using `import` with assets?

File: wallets/rn_cli_wallet_068_5/src/components/CopyURIDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  return (
+    <Dialog.Container
+      visible={visible}
+      blurComponentIOS
+      contentStyle={[styles.mainContainer, {maxWidth: windowWidth * 0.9}]}>
+      <View style={styles.contentContainer}>
+        <Dialog.Title>Enter a WalletConnect URI</Dialog.Title>
+        <View>
+          <Dialog.Description style={styles.descriptionText}>
+            To get the URI press the 📋 copy to clipboard button in wallet
```
Comment (bkrem) [inline wallets/rn_cli_wallet_068_5/src/components/CopyURIDialog.tsx]: Shouldn't this be referencing dapp UIs? E.g. "in a dapp's WalletConnect interface". 

I think "wallet connection interfaces" can be misread as coming from a wallet and cause confusion.

File: wallets/rn_cli_wallet_068_5/src/components/HomeScreen/ActionButtons.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+const ActionButtons = ({setCopyDialog}: IndividualSessionProps) => {
+  return (
+    <View style={styles.absoluteFlexRow}>
+      <CircleActionButton
+        copyImage={true}
+        handlePress={() => {
+          setCopyDialog(true);
+        }}
+      />
+      {/* <CircleActionButton
```
Comment (bkrem) [inline wallets/rn_cli_wallet_068_5/src/components/HomeScreen/ActionButtons.tsx:19]: Needed?

File: wallets/rn_cli_wallet_068_5/src/components/HomeScreen/Sessions.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
+import React, {useEffect} from 'react';
+import {View, Image, Text, StyleSheet, ScrollView} from 'react-native';
+import IndividualSession from './IndividualSession';
+import {web3wallet} from '../../utils/Web3WalletClient';
+
+const Sessions = () => {
+  const sessions = Object.values(web3wallet.getActiveSessions());
+
+  useEffect(() => {}, [sessions]);
```
Comment (bkrem) [inline wallets/rn_cli_wallet_068_5/src/components/HomeScreen/Sessions.tsx]: no-op useEffect?

File: wallets/rn_cli_wallet_068_5/src/components/HomeScreen/Sessions.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+  // @notice: Empty State with no Session
+  if (!sessions || sessions.length === 0) {
+    return (
+      <View style={styles.container}>
+        <Image
+          source={require('../../assets/Tick.png')}
```
Comment (bkrem) [inline wallets/rn_cli_wallet_068_5/src/components/HomeScreen/Sessions.tsx]: I would do the require up front/assign to a const outside of the React component, otherwise this will only throw at runtime if the asset is missing.

File: wallets/rn_cli_wallet_068_5/src/components/HomeScreen/Sessions.tsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+          code that is displayed in the app.
+        </Text>
+      </View>
+    );
+  }
+
+  // @notice: Main Rendering of Sessions
+  return (
+    <ScrollView contentContainerStyle={styles.scrollViewContainer}>
+      {sessions &&
```
Comment (bkrem) [inline wallets/rn_cli_wallet_068_5/src/components/HomeScreen/Sessions.tsx]: Truthy check here is redundant since you're checking the case in the `if` above already 👍 

```suggestion
      {
```

File: wallets/rn_cli_wallet_068_5/src/modals/CopyWCURIModal.tsx [hunk 9]
```diff
@@ -0,0 +1,67 @@
+import React, {useEffect} from 'react';
```
Comment (bkrem) [inline wallets/rn_cli_wallet_068_5/src/modals/CopyWCURIModal.tsx]: Unclear to me atm: why is there a `modals` dir outside of `components`, when there's already e.g. `components/MethodsModal`?

File: wallets/rn_cli_wallet_068_5/src/components/W3WText.tsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+interface IW3WTextProps {
+  value: string;
+  type?: string;
+  color?: string;
+}
+
+/*
+  @notice W3WText Component (Unfinished / not swapped out for all TEXT components)
+  @params value, color
```
Comment (bkrem) [inline wallets/rn_cli_wallet_068_5/src/components/W3WText.tsx]: Unfinished comment (`Color:`), `@params` is misaligned with the current props. 

Adding contextual comments (i.e. answering "why?" questions) is great but I don't think there's a need for full Solidity-style natspec, especially with WIP like this since it will become misaligned/misleading if not kept in shape.

File: wallets/rn_cli_wallet_068_5/src/data/EIP155.tsx [hunk 11]
```diff
@@ -0,0 +1,90 @@
+/**
+ * @desc Refference list of eip155 chains
```
Comment (bkrem) [inline wallets/rn_cli_wallet_068_5/src/data/EIP155.tsx]:
```suggestion
 * @desc Reference list of eip155 chains
```

**Comment by bkrem** [comment]
`wallets/rn_cli_wallet_068_5/src/utils/EIP155Addresses.ts` seems to be an empty file

File: wallets/rn_cli_wallet_068_5/src/components/HomeScreen/ActionButtons.tsx (hunk 5 above)
Comment (crypblizz8) [reply wallets/rn_cli_wallet_068_5/src/components/HomeScreen/ActionButtons.tsx:19]: Placeholder reminder for adding the QR code part.

To not  confuse developers reading the repo, I can remove it.

Comment (bkrem) [reply wallets/rn_cli_wallet_068_5/src/components/HomeScreen/ActionButtons.tsx:19]: It's cool just checking it's not an unwanted leftover

File: wallets/rn_cli_wallet_068_5/src/components/CopyURIDialog.tsx [hunk 12]
```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import {Dimensions, StyleSheet, TextInput, View} from 'react-native';
+import Dialog from 'react-native-dialog';
+import {ConnectButton} from './ConnectButton';
+
+interface copyURIDialogProps {
+  wcURI: string;
+  setWCUri: (arg0: string) => void;
```
Comment (ganchoradkov) [inline wallets/rn_cli_wallet_068_5/src/components/CopyURIDialog.tsx]: nitpick: could we name the arguments a bit more verbose

File: wallets/rn_cli_wallet_068_5/src/components/HomeScreen/IndividualSession.tsx [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+//ToDo: Change to TouchableOpacity and navigate to Session on next Sprint
+const IndividualSession = ({name, icons, url}: IndividualSessionProps) => {
+  return (
+    <View style={styles.sessionContainer}>
+      <View style={styles.flexRow}>
+        {icons ? (
+          <Image
+            source={{uri: icons.toString()}}
```
Comment (ganchoradkov) [inline wallets/rn_cli_wallet_068_5/src/components/HomeScreen/IndividualSession.tsx:17]: Wouldn't this throw an exception if icons is empty?

File: wallets/rn_cli_wallet_068_5/src/modals/PairModal.tsx [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+
+export function PairModal({
+  proposal,
+  visible,
+  open,
+  handleAccept,
+}: PairModalProps) {
+  const name = proposal?.params?.proposer?.metadata?.name;
+  const url = proposal?.params?.proposer?.metadata.url;
+  const methods = proposal?.params?.requiredNamespaces.eip155.methods;
```
Comment (ganchoradkov) [inline wallets/rn_cli_wallet_068_5/src/modals/PairModal.tsx]: Is the modal always expecting `eip155` namespace only?

File: wallets/rn_cli_wallet_068_5/src/screens/HomeScreen.tsx [hunk 15]
```diff
@@ -0,0 +1,289 @@
...
+    },
+    [],
+  );
+
+  useEffect(() => {
+    if (web3wallet) {
+      web3wallet.on('session_proposal', onSessionProposal);
+      web3wallet.on('session_request', onSessionRequest);
+    }
+  }, [
```
Comment (ganchoradkov) [inline wallets/rn_cli_wallet_068_5/src/screens/HomeScreen.tsx:166]: why do the listeners have to be subscribed on each value change?
also consider adding `removeEventListener` as this will reach max listeners very fast
```
return () => {
 web3wallet.removeEventListener('session_proposal', onSessionProposal);
  web3wallet.removeEventListener('session_request', onSessionRequest);
}

File: wallets/rn_cli_wallet_068_5/src/components/CircleActionButton.tsx (hunk 3 above)
Comment (crypblizz8) [reply wallets/rn_cli_wallet_068_5/src/components/CircleActionButton.tsx:11]: `require` is a RN necessity for local files. To make it into CLIPBOARD_IMAGE imports will need to do an export with requires. Going to put in QOL for later

See reference [here](https://stackoverflow.com/questions/38887193/where-to-put-images-in-a-react-native-project).

File: wallets/rn_cli_wallet_068_5/src/components/HomeScreen/IndividualSession.tsx (hunk 13 above)
Comment (crypblizz8) [reply wallets/rn_cli_wallet_068_5/src/components/HomeScreen/IndividualSession.tsx:17]: Returns null

File: wallets/rn_cli_wallet_068_5/src/modals/PairModal.tsx (hunk 14 above)
Comment (crypblizz8) [reply wallets/rn_cli_wallet_068_5/src/modals/PairModal.tsx]: Going to update on next sprint for other nameSpaces. Will leave comments.

File: wallets/rn_cli_wallet_068_5/src/modals/CopyWCURIModal.tsx (hunk 9 above)
Comment (crypblizz8) [reply wallets/rn_cli_wallet_068_5/src/modals/CopyWCURIModal.tsx]: Moved all in `Modal` for the individual components and `Modals for the screen type interface

File: wallets/rn_cli_wallet_068_5/src/screens/HomeScreen.tsx (hunk 15 above)
Comment (crypblizz8) [reply wallets/rn_cli_wallet_068_5/src/screens/HomeScreen.tsx:166]: Switched it to just popup on modal state.
Will refactor later similar to web-examples on revisit

```
 if (approvalModal || copyDialog || signModal || sendTransactionModal) {
      web3wallet.on('session_proposal', onSessionProposal);
      web3wallet.on('session_request', onSessionRequest);
    }
```

**Review by ganchoradkov — APPROVED** [review]
:100: :100:

**Review by bkrem — APPROVED** [review]
